storage/innobase/xtrabackup/test/t/check_for_passwords.sh

```diff
@@ -11,37 +11,26 @@ vlog "Preparing server"
 start_server
 mysql -e "CREATE USER pxb@localhost IDENTIFIED BY '$password_string'"
 mysql -e "GRANT ALL ON *.* TO pxb@localhost "
+mysql -e "CREATE TABLE t1(i int) engine=MYISAM" test
+mysql -e "INSERT INTO t1 values(1)" test
 mkdir -p $topdir/backup
 
 #pass option like password -p to be run with 
 function grep_in_datadir_ps() {
-	additional_options=$1
+  additional_options=$1
+  mysql -e "SELECT SLEEP(15) from t1 for update" test &
   $XB_BIN $XB_ARGS --backup -u pxb $additional_options\
   --transition-key=$password_string --target-dir=$topdir/backup \
-  --debug-sync="data_copy_thread_func"  2>&1 | tee $topdir/pxb.log &
-
+  2>&1 | tee $topdir/pxb.log &
   job_pid=$!
-  pid_file=$topdir/backup/xtrabackup_debug_sync
-  # Wait for xtrabackup to suspend
-  i=0
-  while [ ! -r "$pid_file" ]
-  do
-      sleep 1
-      i=$((i+1))
-      echo "Waited $i seconds for $pid_file to be created"
-  done
-  xb_pid=`cat $pid_file`
-
+  #sleep sufficient time to let backup run
+  sleep 5
```

Comment:
Same here, sleep won't be reliable. I would rather either grep the output file and check for a string that indicates the process has started, or keep running a kill -0 on pxb PID.